pkg/bgpv1/manager/reconcilerv2/policies.go

```diff
@@ -0,0 +1,316 @@
+			toRemove = append(toRemove, p)
+		}
+	}
+
+	// tracks which peers have to be reset because of policy change
+	resetPeers := sets.New[string]()
+
+	// add missing policies
+	for _, p := range toAdd {
+		rp.Logger.WithFields(logrus.Fields{
+			types.PolicyLogField: p.Name,
+			"statements":         p.Statements,
+		}).Debug("Adding route policy")
+
+		err := rp.Instance.Router.AddRoutePolicy(rp.Ctx, types.RoutePolicyRequest{
+			DefaultExportAction: types.RoutePolicyActionReject, // do not advertise routes by default
+			Policy:              p,
+		})
+		if err != nil {
+			return runningPolicies, err
+		}
+
+		runningPolicies[p.Name] = p
+		resetPeers.Insert(peerAddressFromPolicy(p))
+	}
+
+	// update modified policies
+	for _, p := range toUpdate {
+		// As proper implementation of an update operation for complex policies would be quite involved,
+		// we resort to recreating the policies that need an update here.
+		rp.Logger.WithFields(logrus.Fields{
+			types.PolicyLogField: p.Name,
+			"statements":         p.Statements,
+		}).Debug("Updating (re-creating) route policy")
+
+		existing := rp.CurrentPolicies[p.Name]
+		err := rp.Instance.Router.RemoveRoutePolicy(rp.Ctx, types.RoutePolicyRequest{Policy: existing})
+		if err != nil {
+			return runningPolicies, err
+		}
+		delete(runningPolicies, existing.Name)
+
+		err = rp.Instance.Router.AddRoutePolicy(rp.Ctx, types.RoutePolicyRequest{
+			DefaultExportAction: types.RoutePolicyActionReject, // do not advertise routes by default
+			Policy:              p,
+		})
+		if err != nil {
+			return runningPolicies, err
+		}
+
+		runningPolicies[p.Name] = p
+		resetPeers.Insert(peerAddressFromPolicy(p))
+	}
+
+	// remove old policies
+	for _, p := range toRemove {
+		rp.Logger.WithFields(logrus.Fields{
+			types.PolicyLogField: p.Name,
+			"statements":         p.Statements,
+		}).Debug("Removing route policy")
+
+		err := rp.Instance.Router.RemoveRoutePolicy(rp.Ctx, types.RoutePolicyRequest{Policy: p})
+		if err != nil {
+			return runningPolicies, err
+		}
+		delete(runningPolicies, p.Name)
+		resetPeers.Insert(peerAddressFromPolicy(p))
+	}
+
+	// soft-reset affected BGP peers to apply the changes on already advertised routes
+	for peer := range resetPeers {
+		_, err := netip.ParsePrefix(peer)
+		if err != nil {
+			continue
+		}
+
+		rp.Logger.WithFields(logrus.Fields{
+			types.PeerLogField: peer,
+		}).Debug("Resetting peer due to a routing policy change")
+
+		req := types.ResetNeighborRequest{
+			PeerAddress:        peer,
+			Soft:               true,
+			SoftResetDirection: types.SoftResetDirectionOut, // we are using only export policies
+		}
+
+		err = rp.Instance.Router.ResetNeighbor(rp.Ctx, req)
+		if err != nil {
+			// non-fatal error (may happen if the neighbor is not up), just log it
+			rp.Logger.WithFields(logrus.Fields{
+				types.PeerLogField: peer,
+			}).WithError(err).Warn("error by resetting peer after a routing policy change")
```

Comment:
We don't see this error on peer delete, since neighbor reconciler has lower priority compared to advertisement reconcilers. So all paths, policies gets configured before peer is configured/removed.

I modified it to be debug log anyways, I agree error is bit noisy.